modules/sievefilters/modules.php

```diff
@@ -194,9 +194,12 @@ function get_blocked_senders($mailbox, $mailbox_id, $icon_svg, $icon_block_domai
         return '';
     }
     $current_script = $client->getScript('blocked_senders');
+    $blocked_list_actions = [];
     if ($current_script != '') {
         $base64_obj = str_replace("# ", "", preg_split('#\r?\n#', $current_script, 0)[1]);
         $blocked_list = json_decode(base64_decode($base64_obj));
+        $base64_obj_actions = str_replace("# ", "", preg_split('#\r?\n#', $current_script, 0)[2]);
```

Comment:
There is a possibility that $current_script is not in the format you expect, so preg_split()[] will fail. Also, this line is repeated two times. Can you make it more robust by splitting first, then checking if the format is OK (split result contains 2 items) and then proceed to use them?

Reply:
Current script has always this format. Sometimes line 2 is empty.
`
# CYPHT CONFIG HEADER - DON'T REMOVE
# PyIqQCpnioqwerbWFbC5jb20iLCIqQG1hcUoiN6m9YXBvcnRlLmNvbSJd
# eyIqQG1hcmNsYXBvcnRlLmNvbSI6eyJhY3Rpb24iOiJyZWplY3RfZGVmYX

# Requirements
require ["regex","reject"];

if header :contains "From" ["*@*xyz.com"] {
	discard;
	stop;
}
`